### PR TITLE
Do not return err on symlink eval

### LINKED ISSUE #10146: bind mount mkdir on create vs start is fun with folders in tmpfs
So, I have an apt-cacher-ng container that has a bind mount of `-v /tmp/apt-cacher-ng:...` (and `/tmp` on my host is `tmpfs`).  When I `docker run` it, I use `--restart=always`, and it starts up swell (because `/create` does a `mkdir`).

Now, I reboot my machine and my apt-cacher-ng is gone and fails to start up unless I manually `mkdir /tmp/apt-cacher-ng` because we don't `mkdir` on `/start`, which is certainly confusing the first time you see it because `-v` generally does `mkdir` for bind mounts. :smile:

cc @cpuguy83

### PR DESCRIPTION
**Pull request by cpuguy83**
Fixes #10146 as a nice side effect as well.

### CODE REVIEW

**Comment by cpuguy83** [comment]
Currently the only way to hit this error is to remove the block of code in daemon/start.go#setHostConfig which creates the paths for binds.
What's weird is the error returned when doing this is an `Lstat: No such file or directory`, but docker run/docker create gives either:

1. If --name is specified, says that --name is already taken and it's not
2. If no --name, it says it can't find the image

**Comment by cpuguy83** [comment]
Also, this could potentially be hit if someone fires up a daemon and there is a container which has a bind mount set for a host path which does not exist.

I'm going to add some unit tests for volumes separately.

**Comment by icecrime** [comment]
Ping @crosbymichael

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -403,3 +403,77 @@ func TestDaemonKeyMigration(t *testing.T) {
...
+	graphDir := filepath.Join(os.TempDir(), "docker-test")
+	defer os.RemoveAll(graphDir)
+	if err := d.StartWithBusybox("-g", graphDir); err != nil {
+		t.Fatal(err)
+	}
+
+	tmpDir := filepath.Join(os.TempDir(), "test")
+	defer os.RemoveAll(tmpDir)
+
+	if out, err := d.Cmd("create", "-v", fmt.Sprintf("%s:/foo", tmpDir), "--name=test", "busybox"); err != nil {
```
Comment (tiborvass) [inline integration-cli/docker_cli_daemon_test.go]: @cpuguy83 no need for Sprintf if you only have one variable: `tmpDir + ":/foo"` is sufficient.

Comment (cpuguy83) [inline integration-cli/docker_cli_daemon_test.go]: Updated

**Comment by tiborvass** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**